core/controllers/profile_test.py

```diff
@@ -1168,7 +1168,7 @@ def raise_exception() -> None:
                     'email': 'email@example.com',
                     'tag': 'Web',
                     'name': 'Name'
-                }, csrf_token=csrf_token, expected_status_int=500)
+                }, csrf_token=csrf_token, expected_status_int=405)
```

Comment:
Should we be changing the status code? The '/mailinglistsubscriptionhandler' do accepts the 'put' requests, right?

Reply:
Hey @lkbhitesh07, so the exception raised for this test case is handled by the `handle_exception()` method of core/controllers/base.py, specifically this if block:
```python
if isinstance(exception, TypeError):
            self.error(405)
            values = {
                'error': 'Invalid method %s for %s/ Extra Parameters passed' % (
                    request_method, handler_class_name),
                'status_code': 405
            }
            self._render_exception(values)
            return
``` 
I tried to add another if block inside this one to check if the request method is in the list of methods supported by the handler, which would solve the breaking of this test, but that breaks the code and gives a 500 error for all cases (even the ones where we expect a 405).  Due to this, I replaced the 500 with 405. I also changed the error message of the exception to say 'Invalid method or Extra parameters passed' specifically for this case as I believe the 'name' parameter passed here:
```
{
                    'email': 'email@example.com',
                    'tag': 'Web',
                    'name': 'Name'
                }
``` 
 is extra. This test seems to be an edge case as the changes I have made work for all the other routes and tests.

Reply:
Got it, thanks @mon4our